lib/openhab/dsl/items/generic_item.rb

```diff
@@ -131,6 +131,23 @@ def groups
           group_names.map { |name| Groups.groups[name] }
         end
 
+        # Return the item's thing if this item is associated with a thing. If an item is bound to more than one thing,
+        # this method only returns the first thing.
+        #
+        # @return [Thing] The thing associated with this item or nil
+        def thing
+          all_things.first
+        end
+
+        # Returns all of the item's associated things.
+        #
+        # @return [Array] An array of things or an empty array
+        def all_things
```

Comment:
I had to name this all_things instead of the preferred things. I was getting a namespace collision with things[] that managed to put my openHAB instance into a bad state. I would love a better name suggetion

Reply:
Hmm. I know we're probably leaking `Object.things` onto too many things, but when you define it on GenericItem it should be hiding the inherited method.

Reply:
That's probably what the problem was. Items were calling this things method when they really wanted Object.things and odd things were happening.

I had though maybe "all_linked_things" to really separate it out.

Reply:
Ooh I like that. `linked_thing` and `all_linked_things` for the method names. Definitely gives these methods a closer association with how links show up in the UI.